src/main/java/org/marketdesignresearch/mechlib/core/bidder/valuefunction/XORValueFunction.java

```diff
@@ -48,7 +48,11 @@ public Bid toBid(UnaryOperator<BigDecimal> bundleBidOperator) {
     }
 
     public List<BundleValue> getOptimalBundleValueAt(Prices prices, int maxNumberOfBundles) {
-        return bundleValues.stream()
+        return this.getOptimalBundleValueAt(prices, maxNumberOfBundles, false);
+    }
+    
+    public List<BundleValue> getOptimalBundleValueAt(Prices prices, int maxNumberOfBundles, boolean allowNegative) {
+        return bundleValues.stream().filter(b -> allowNegative || b.getAmount().subtract(prices.getPrice(b.getBundle()).getAmount()).signum() >= 0)
                 .sorted((a, b) -> {
                     BigDecimal first = a.getAmount().subtract(prices.getPrice(a.getBundle()).getAmount());
                     BigDecimal second = b.getAmount().subtract(prices.getPrice(b.getBundle()).getAmount());
```

Comment:
With this new function it is still possible to get bundles with a negative utility. Note that unlike the original filtering in the XORBidder it also returns bundles with a utility of 0 (which is not negative).